src/postprocessor/compress.rs

```diff
@@ -49,20 +110,42 @@ impl Postprocessor for Zlib {
     }
 }
 
-#[derive(Default)]
-struct Xz2 {}
+struct Xz2 {
+    compression_level: u32,
+}
+impl Xz2 {
+    fn with_config(level: i64) -> Result<Box<Self>> {
+        if !(0..=9).contains(&level) {
+            return Err(format!(
+                "Xz2 supports compression level between 0 and 9 but {level} was given"
+            )
+            .into());
+        }
+
+        Ok(Box::new(Self {
+            compression_level: level.try_into().expect("could not parse compression level"),
```

Comment:
expect always panics, lets just return an `Err` instead:

```suggestion
            compression_level: level.try_into()?,
```